File: MadeByDade.Living.Data/Bills/BillPayment.cs
using System.ComponentModel.DataAnnotations;

namespace MadeByDade.Living.Data.Bills;

public class BillPayment
{
    public int Id { get; set; }
    public DateTime DateDue { get; set; }
    public DateTime? DatePaid { get; set; }
    public DateTime CreatedOn { get; set; } = DateTime.Today;

    public int BillId { get; set; }
    public Bill Bill { get; set; }

    public bool IsPaid => DatePaid.HasValue;
}

File: MadeByDade.Living.React/src/bills/BillPayment.ts
import { z } from "zod";
import { BillSchema } from "./Bill";

export const BillPaymentSchema = z.object({
	id: z.number(),
	billId: z.number(),
	dateDue: z.string(),
	datePaid: z.string().nullable(),
	createdOn: z.string(),
	isPaid: z.boolean(),

	bill: BillSchema,
});

export const BillPaymentsSchema = z.array(BillPaymentSchema);

export type BillPayment = z.infer<typeof BillPaymentSchema>;
